pkg/cdi/cdi.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cdi
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	cdiSpecs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"github.com/golang/glog"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+const cdiSpecPrefix = "sriov-dp-"
+
+// CDI represents CDI API required by Device plugin
+type CDI interface {
+	CreateCDISpecForPool(resourcePrefix string, rPool types.ResourcePool) error
+	CreateContainerAnnotations(devicesIDs []string, resourcePrefix, resourceKind string) (map[string]string, error)
+	CleanupSpecs() error
+}
+
+// impl implements CDI interface
+type impl struct {
+}
+
+// New returns an instance of CDI interface implementation
+func New() CDI {
+	return &impl{}
+}
+
+// CreateCDISpecForPool creates CDI spec file with specified devices
+func (c *impl) CreateCDISpecForPool(resourcePrefix string, rPool types.ResourcePool) error {
+	err := c.CleanupSpecs()
+	if err != nil {
+		glog.Error("CreateCDISpecForPool(): can not cleanup old spec files", err)
```

Comment:
Shouldn't this be `glog.Errorf("....%v", err)`?

Reply:
fixed